src/models/OtrStore.js

```diff
@@ -0,0 +1,67 @@
+var _ = require("lodash");
+var DSA = require('otr').DSA;
+var OTR = require('otr').OTR;
+var Chan = require("./chan");
+
+module.exports = OtrStore;
+
+/** Stores OTR state for a given user
+ *
+ * Handels private key, current sessions and fingerprints.
+ */
+function OtrStore(name, client) {
+	_.merge(this, {
+		name: name,
+		privKey: undefined,
+		sessions: {},
+		client: client,
+	});
+	console.log('Generating DSA....');
+	this.privKey = new DSA();
+	console.log('OK, key generated');
+}
+
+/** Initialize an OTR session for query chans
+ *
+ * Event handler, to be called on chan creation
+ */
+OtrStore.prototype.initChan = function(args) {
+	var network_id = args.network;
+	var chan = args.chan;
+	if (chan.type == Chan.Type.QUERY) {
+		var network = _.find(this.client.networks, {id: network_id});
+		var session = this.getSession(chan.name, network);
+		var irc = network.irc;
+		if (session == undefined) {
+			session = new OTR({
+				fragment_size: 140,
+				send_interval: 200,
```

Comment:
Sorry, maybe I didn't explain myself properly. Reading the code as someone not involved in this file, it doesn't strike me as obvious. Without writing all of the lib doc here, understanding the code at first glance might vital here, especially if someone reviews the final code to merge it with the upstream and doesn't want to dig into the lib.

But it's okay, we have more pressing concerns at the moment. I'd like to do an entire pass of documenting the overall code myself anyway, before sending the full PR.

:+1: